### ISSUE TITLE
IDE0059 is triggered for recurrent lambda initialization

### ISSUE THREAD

**Issue by premun** [dotnet/roslyn#48622]
**Version Used**: 
VS 16.9.0 Preview 1.0 [30609.169.main]

**Steps to Reproduce**:

```csharp
using System;

namespace TestNamespace
{
    public class TestClass
    {
        public void TestMethod()
        {
            Func<string, string> func = null;
            func = s =>
            {
                // When initialization above is removed, this line results in an error:
                // Use of unassigned local variable 'func'
                return func(s);
            };
        }
    }
}
```

**Expected Behavior**:
IDE0059 should not be triggered for `func`.

**Actual Behavior**:
IDE0059 is triggered for `func` but when initialization to `null` is removed, compiler raises error:
> **Use of unassigned local variable 'func'**

For the `return func(s);` line.
